qadence2_expressions/core/expression.py

```diff
@@ -558,11 +559,14 @@ def __kron__(self, other: object) -> Expression:
             return self
 
         # ⚠️ Warning: Ideally, this step should not perform the evaluation of the
-        # the expression. However, we want to provide a friendly intercation to
-        # the users, and the inacessibility of Python's evaluation (without writing
-        # our on REPL) forces to add the evaluation at this point.
+        # expression. However, we want to provide a friendly interaction to the users,
+        # and the inaccessibility of Python's evaluation (without writing our own REPL)
+        # forces to add the evaluation at this point.
         return evaluate_kron(Expression.kron(self, other))
 
+    def __matmul__(self, other: object) -> Expression:
+        return self.__kron__(other)
```

Comment:
The reason it wasn't exposed initially was to avoid confusion on how multiplication works.

Having both options, `X(1) * Z(2)` and `X(1) @ Z(2)`, adds an unnecessary level of complexity from the use point of view.

Reply:
I would also think so. Maybe this is needed for backwards compatibility purposes ?